### PR TITLE
homework_5

### PR DESCRIPTION
**Pull request by IuriiSky**

### CODE REVIEW

File: homeworks/iurii.stepanenko_IuriiSky/homework_5/task2.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+function returnKey(params) {
+    const obj = {};
+    params.forEach((element) => {
+        if (obj[element] === undefined) {
+            Object.defineProperty(obj, element, {
+                enumerable: true,
+                configurable: false,
+                writable: true,
+                value: 1,
```
Comment (markoffden) [inline homeworks/iurii.stepanenko_IuriiSky/homework_5/task2.js]: very cool, but you should not go that deep without a need

File: homeworks/iurii.stepanenko_IuriiSky/homework_5/task3.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const obj = { b: 'c', d: { e: 'f' } };
+function innerCopy(newObj) {
+    return Object.assign({}, newObj);
+}
```
Comment (markoffden) [inline homeworks/iurii.stepanenko_IuriiSky/homework_5/task3.js]: this is not a deep copy as objects `d` property still reference the same object

File: homeworks/iurii.stepanenko_IuriiSky/homework_5/task4.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    if (targetPerson === undefined) {
+        return null;
+    }
+    const friendsPeople = [];
+    if (targetPerson.friends === null) {
+        return friendsPeople;
+    }
+    targetPerson.friends.forEach((element) => {
+        const friendPerson = people.find(person => person.id === element);
+        friendsPeople.push(friendPerson);
```
Comment (markoffden) [inline homeworks/iurii.stepanenko_IuriiSky/homework_5/task4.js]: here you could just use `filter()` for example

Comment (IuriiSky) [reply homeworks/iurii.stepanenko_IuriiSky/homework_5/task4.js]: I fix (change), maybe this code looks better?

**Comment by markoffden** [comment]
check the object copy task, it's wrong

**Comment by IuriiSky** [comment]
I'm working on this, maybe give me a hint? =)

**Comment by markoffden** [comment]
when you assign a primitive value like `string`, `number` or `boolean` you assign a value

when you assign a complex value (`object`, `array`) you assign a reference to value

that's why when you copy an `object` type property simply by `=` it's copying the reference to the same object

you should check if the property is object and if it is run you function recursively

File: homeworks/iurii.stepanenko_IuriiSky/homework_5/task3.js (hunk 2 above)
Comment (IuriiSky) [reply homeworks/iurii.stepanenko_IuriiSky/homework_5/task3.js]: I fix this task. I hope it`s right.

**Pull request merged**